src/main/java/org/prebid/server/execution/RemoteFileSyncer.java

```diff
@@ -26,59 +28,43 @@
 import java.util.Objects;
 import java.util.concurrent.TimeoutException;
 
-/**
- * Works with remote web resource.
- */
 public class RemoteFileSyncer {
 
     private static final Logger logger = LoggerFactory.getLogger(RemoteFileSyncer.class);
 
-    private final String downloadUrl;  // url to resource to be downloaded
-    private final String saveFilePath; // full path on file system where downloaded file located
-    private final String tmpFilePath; // full path on file system where tmp file located
-    private final int retryCount; // how many times try to download
-    private final long retryInterval; // how long to wait between failed retries
+    private final String downloadUrl;
+    private final String saveFilePath;
+    private final String tmpFilePath;
+    private final RetryPolicy retryPolicy;
     private final long timeout;
     private final long updatePeriod;
     private final HttpClient httpClient;
     private final Vertx vertx;
     private final FileSystem fileSystem;
 
-    private RemoteFileSyncer(String downloadUrl, String saveFilePath, String tmpFilePath, int retryCount,
-                             long retryInterval, long timeout, long updatePeriod, HttpClient httpClient, Vertx vertx,
-                             FileSystem fileSystem) {
-        this.downloadUrl = downloadUrl;
-        this.saveFilePath = saveFilePath;
-        this.tmpFilePath = tmpFilePath;
-        this.retryCount = retryCount;
-        this.retryInterval = retryInterval;
+    public RemoteFileSyncer(String downloadUrl,
+                            String saveFilePath,
+                            String tmpFilePath,
+                            RetryPolicy retryPolicy,
+                            long timeout,
+                            long updatePeriod,
+                            HttpClient httpClient,
+                            Vertx vertx) {
+
+        this.downloadUrl = HttpUtil.validateUrl(downloadUrl);
+        this.saveFilePath = Objects.requireNonNull(saveFilePath);
+        this.tmpFilePath = Objects.requireNonNull(tmpFilePath);
+        this.retryPolicy = Objects.requireNonNull(retryPolicy);
```

Comment:
Is it possible when we use `basicGeoLocationService` for this properties to be nulls(and not having exceptions is the intended behavior)?

Reply:
No.
`download url`, `save file path` and `temp file path` are needed to actually download something.
`retry policy` also cannot be null, because even if we don't want to use retries we should pass NoRetryPolicy instance.